### PR TITLE
Check private google access enabled

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
Whats not to like?

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class GoogleSubnetworkPrivateGoogleEnabled(BaseResourceValueCheck):
+    def __init__(self) -> None:
+        name = "Ensure that private_ip_google_access is enabled for Subnet"
+        id = "CKV_GCP_74"
+        supported_resources = ("google_compute_subnetwork",)
+        categories = (CheckCategories.GENERAL_SECURITY,)
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def get_inspected_key(self) -> str:
+        return "private_ip_google_access"
```
[the remark this answers is not in the document]
Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py:17]: would it harm to have both? Can we do both and have autoremediation? or have another check that that looks for ipv6 use and checks for that attribute?

[the remark this answers is not in the document]
Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py:17]: ok expect a new pr later that - private_ipv6_google_access is true for ipv6 networks

[the remark this answers is not in the document]
Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py:17]: Its mate is in the pr pipe.

**Review by gruebel — APPROVED** [review]
🍸

**Pull request merged**